src/main/java/com/coveo/AbstractFMT.java

```diff
@@ -115,19 +120,23 @@ protected void postExecute(List<String> filesProcessed, int nonComplyingFiles)
     return filesProcessed;
   }
 
-  private void formatSourceFilesInDirectory(File directory) throws MojoFailureException {
+  public void formatSourceFilesInDirectory(File directory, Formatter formatter)
+      throws MojoFailureException {
     if (!directory.isDirectory()) {
       logger.info("Directory '" + directory + "' is not a directory. Skipping.");
       return;
     }
 
-    List<File> files = Arrays.asList(directory.listFiles(getFileFilter()));
-    for (File file : files) {
-      if (file.isDirectory()) {
-        formatSourceFilesInDirectory(file);
-      } else {
-        formatSourceFile(file);
-      }
+    try (Stream<Path> paths = Files.walk(Paths.get(directory.getPath()))) {
+      paths
+          .collect(Collectors.toList())
+          .parallelStream()
```

Comment:
Using this approach you'll get a better performance gain if you have a single folder with many files vs multiple folders with few files.

I would suggest flatmap, as proposed by @malaporte 

Also, I would recommend using an approach that allows specifying a thread pool / executor. Is the default one use by `parallelStream` sufficient here?

Reply:
I agree that the vanilla parallelStream might not be optimal, but it does provide some improvement. I'll leave it as future work to tweek the necessary parameters to get the best performances out of it.